Fix job URL when Jenkins URL is set and address URL encoding when job URLs are already quoted

dloucasfx opened the pull request:
When a Jenkins instance is only configured with `--prefix` and the monitor setting `path` is set, the `path` value is appended to the job URL correctly, but when `Jenkins URL` is configured in addition to `--prefix` the Job URL will already contain the path (assuming prefix and Jenkins URL path are identical).

This PR checks the start of the Job path to see if the monitor `path` already exists or needs to be added.

While validating this fix, I have noticed that the job URLs fetched from Jenkins are already encoded and the `_api_call` was failing as it does a second `quote` (changing the correct path)
I assume this is a recent change from Jenkins, to keep it backward compatible, we are now unquoting the job URL before calling  `_api_call`

cc: @keitwb @dtcimbal 
Signed-off-by: Dani Louca <dlouca@splunk.com>

rmfitzpatrick commented on jenkins.py:
```diff
@@ -549,7 +550,13 @@ def read_metrics(module_config):
+            if module_config["path"] and job_path.startswith(module_config["path"]):
+                job_url = urllib_unquote(job["url"])
+            else:
+                job_url = module_config["base_url"][:-1] + urllib_unquote(urllib.parse.urlparse(job["url"]).path)
```
```suggestion
                job_url = module_config["base_url"][:-1] + urllib_unquote(job_path)
```

dloucasfx replied:
oh... that's an oversight 🤦  good catch!

rmfitzpatrick commented:
Can you add an updated sample response and unit test to demonstrate the issue?

dloucasfx commented:
> Can you add an updated sample response and unit test to demonstrate the issue?

Fixed the integration and added a sample job that shows an encoded job name.
Note: none of the existing test suite will directly validate the fix I made here, but I validated that both mock and integration still passes
